src/java.base/share/classes/sun/security/ssl/SignatureScheme.java

```diff
@@ -437,8 +434,10 @@ private boolean isPermitted(AlgorithmConstraints constraints) {
                 }
             } else if (ss.isAvailable &&
                     ss.supportedProtocols.contains(protocolVersion) &&
-                    (config.signatureSchemes.isEmpty() ||
-                        config.signatureSchemes.contains(ss)) &&
+                    (config.signatureSchemes == null ||
+                        config.signatureSchemes.length == 0 ||
+                        Arrays.asList(config.signatureSchemes)
```

Comment:
Is that a allocation per handshake*availablesize?

Reply:
Yes, it does.  Do you like to use a for-loop, without new object allocation?

Reply:
Yes, static helper

Reply:
Thanks for confirm.  Updated to use a static helper method.